packages/devtools_app/lib/src/primitives/notifications.dart

```diff
@@ -0,0 +1,56 @@
+// Copyright 2022 The Chromium Authors. All rights reserved.
```

Comment:
what is the benefit of splitting the notifications up into multiple libraries? When working in the codebase, I would find it a bit confusing that all the notifications code is not in the same place. We don't do any deferred loading or anything like that, so it is not a problem for us if files in other areas of DevTools need to import something from `framework/`

Reply:
There are two parts: interface and implementation.
Interface is used by screens and sits in `shared`. 
Implementation sits in `framework` and is used only by App. 

It is much easier to understand interface when it is clearly separated. Now shared/notifications.dart contains three classes intended to be used by screens. It is easy to observe and understand them. If implementation code was here, it would be harder.

Renamed implementation to notifications_view.dart to make it clearer.

Reply:
> The fact that implementation sits in the folder framework that is forbidden for screens, creates some level of protection.

A screen can still import something from `framework/` - there is no rule against that. Can you elaborate on the "protection" you are referring to? What are you concerned about guarding against?

Reply:
It is just mental agreement about content of 'framework'. No formal protection.
We put things intended for screens to 'sharing', and things not intended for screens, to 'framework'.
For example, if some screen decides to host NotificationsView, things will mess up and may not be caught by a test. 
And we always can change our mind about what we want to share and move this thing from 'framework' to 'shared'.